components/User/UserProfile/Integrations.js

```diff
@@ -0,0 +1,390 @@
+            //do nothing
+            return;
+        }
+
+        this.provider = e.target.attributes[1].nodeValue;
+
+        this.context.executeAction(updateProviderAction, 'enable_' + this.provider);
+
+        //prepare localStorage
+        localStorage.setItem(MODI, 'addProvider');
+        localStorage.setItem(NAME, '');
+
+        //observe storage
+        $(window).off('storage').on('storage', this.handleStorageEvent.bind(this));
+
+        //create new tab
+        let url = 'http://authorizationservice.manfredfris.ch:3000/connect/' + this.provider;
+
+        let width = screen.width*0.75, height = screen.height*0.75;
+        if (width < 600)
+            width = screen.width;
+        if (height < 500)
+            height = screen.height;
+        let left = screen.width/2-width/2, topSpace = screen.height/2-height/2;
+
+        let win = window.open(url, '_blank', 'width='+width+',height='+height+',left='+left+',top='+topSpace+',toolbar=No,location=No,scrollbars=no,status=No,resizable=no,fullscreen=No');
+        win.focus();
+    }
+
+    handleDisable(e) {
+        console.log('handleDisable', e.target.attributes[1].nodeValue);
+        e.preventDefault();
+
+        if (this.props.UserProfileStore.providerAction !== '') {
+            //do nothing
+            return;
+        }
+
+        this.provider = e.target.attributes[1].nodeValue;
+
+        this.context.executeAction(updateProviderAction, 'disable_' + this.provider);
+
+        if (this.props.UserProfileStore.user.providers.length === 1 && this.props.UserProfileStore.user.hasPassword === false) {
+            swal({
+                title: 'Error',
+                text: 'You are not allowed to disable all providers.',
+                type: 'error',
+                confirmButtonText: 'Confirmed',
+                confirmButtonClass: 'negative ui button',
+                buttonsStyling: false
+            }).then().catch();
+
+            //just stop here
+            this.context.executeAction(updateProviderAction, '');
+            return;
+        }
+
+        //TODO show spinner
+
+
+        this.context.executeAction(removeProvider, this.provider);
+    }
+
+    handleStorageEvent(e) {
+        console.log('storage event', e.key, localStorage.getItem(e.key));
+        //this is available
+
+        if (e.key !== NAME || localStorage.getItem(MODI) !== 'addProvider')
+            return false;
+
+        let data = {};
+        try {
+            data = JSON.parse(localStorage.getItem(e.key));
+        } catch (err) {
+            console.log('Error while parsing data', err);
+            this.context.executeAction(updateProviderAction, '');
+            return false;
+        }
+        finally {
+            //delete data
+            localStorage.setItem(NAME, '');
+        }
+
+        //add language before send to service
+        let language = navigator.browserLanguage || navigator.language;
```

Comment:
The whole paragraph seems to be repeated in another file --> export into a function --> avoids copy and paste